stl/inc/yvals_core.h

```diff
@@ -2034,5 +2043,24 @@ compiler option, or define _ALLOW_RTCc_IN_STL to suppress this error.
 #define _RESTRICT __restrict
 #endif // ^^^ !defined(__CUDACC__) ^^^
 
+#if _HAS_CXX20 // see GH-5225
+#if defined(__EDG__)
+#define _HAS_IF_CONSTEVAL _HAS_CXX23
+#elif defined(__clang__) || !defined(__CUDACC__) // ^^^ EDG / Clang or MSVC vvv
+#define _HAS_IF_CONSTEVAL 1
+#else // ^^^ Clang or MSVC / others vvv
+#define _HAS_IF_CONSTEVAL defined(__cpp_if_consteval)
+#endif // ^^^ others ^^^
+
+#if _HAS_IF_CONSTEVAL
+#define _STL_CONSTEVAL_CONDITION consteval

```

Comment:
This is mixing together two significant changes - the bugfix for destroying objects, and using `if consteval` in headers. This adds risk and makes it more difficult to review each change.

Can you create separate PRs for these changes?

Reply:
Yeah. The changes about `if consteval` are dropped from this PR now.